articles/create-a-node-cli/index.md

```diff
@@ -0,0 +1,257 @@
+
+## The Shebang
+
+First things first, the starting point of any CLI application point is a line called the Shebang.
+In Node.js the shebang looks something like this:
+
+```javascript
+#!/usr/bin/env node
+
+```
+
+The shebang tells the operating system what type of code is in the script. It also tells which interpreter it should assign to the script. If it were a python script, it would look something like this:
+
+```python
+#!/usr/bin/env py
+```
+
+The shebang, is only applicable in Unix/Linux operating systems.
+It should always be the first line of the entry file.
+
+### Project Setup
+
+To get started, let's setup our project and Node.js environment. In the terminal, enter:
+
+```terminal
+mkdir node-cli
+cd node-cli
+npm init -y
+```
+
+NOTE: passing the `-y` flag to npm init, will instantiate a new npm project and set everything to default and yes.
+
+This will generate our package.json file.
+
+Next up let's install all the packages we will make use of in the project.
+We will make use of the following:
+
+- [axios](https://www.npmjs.com/package/axios) for making our network requests.
+- [chalk](https://www.npmjs.com/package/chalk) for making our user interface better looking and stylish.
+- [yargs](https://www.npmjs.com/package/yargs) for all our options and argument parsing.
+
+Let's go ahead and install them:
+
+```terminal
+    npm i axios chalk yargs
+```
+
+### Getting the jokes and quotes
+
+Let's setup the jokes and quotes. We will be using a [quote-generator-api](https://quotes.rest/) and [jokes-generator-api](https://official-joke-api.appspot.com/). We will fetch the individual jokes and display them on the terminal when the user asks for them.
+
+We will create two functions that will do the fetching using axios.
+
+```javascript
+#!/usr/bin/env node
+
+const axios = require("axios");
+const chalk = require("chalk");
+
+let url = "";
+
+function getJoke() {
+  url = "https://official-joke-api.appspot.com/random_joke";
+ axios({
+ method: "get",
+ url: url,
+ })
+ .then((res) => {
+ const setup = chalk.cyan(res.data.setup);
+ const punchline = chalk.green(res.data.punchline);
+ console.log(`${setup} - ${punchline}`);
+ })
+ .catch((err) => {
+ const log = chalk.red(err);
+ console.log(log);
+ });
+}
+
+function getQuote() {
+axios({
+    method: 'get',
+    url: "http://quotes.stormconsultancy.co.uk/random.json",
+}).then((res)=> {
+    const randomQuote = chalk.cyan(res.data.quote);
+    const randomAuthor = chalk.green(res.data.author);
+    const log = `${randomQuote} - ${randomAuthor}`;
+    console.log(log);
+}).catch((err)=>{
+    const errorLog = chalk.red(err);
+    console.log(errorLog);
+})
+```
+
+Note: We are using chalk to color our logs to get a nice feel and a colorful output. We setup an empty string called url so that when the user selects quotes, we load in the quote api url and same goes for jokes.
```

Comment:
"color our logs and get a colorful output" would flow better.

and **the** same goes for jokes